packages/profile-sync-controller/src/controllers/user-storage/UserStorageController.ts

```diff
@@ -810,7 +801,11 @@ export default class UserStorageController extends BaseController<
           continue;
         }
       }
+
+      this.#accounts.isAccountSyncingInProgress = false;
     } catch (e) {
+      this.#accounts.isAccountSyncingInProgress = false;
```

Comment:
Nice coverage in the catch statement.

Nit/comment (doesn't need to be actionable) - Can we use `finally` statement to finish  account syncing? I am unsure if `finally` is triggered when errors are re-thrown in a `catch` block.

Reply:
@Prithpal-Sooriya nice, I'll add that! I always forget about finally, damn...

From MDN: 

Control flow will always enter the finally block, which can proceed in one of the following ways:

- Immediately after the try block finishes execution normally (and no exceptions were thrown);
- Immediately after the catch block finishes execution normally;
- Immediately before the execution of a control-flow statement (return, throw, break, continue) in the try block or catch block that would exit the block.

So yes, finally should cover our case even if it throws in the catch block.
On it!